src/main/scala/soc/ChipIdPin.scala

```diff
@@ -0,0 +1,44 @@
+package testchipip.soc
+
+import chisel3._
+import chisel3.util._
+import org.chipsalliance.cde.config._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.regmapper._
+import freechips.rocketchip.subsystem._
+
+case class ChipIdPinParams (
+  numChips: Int = 1, 
+  chipIdAddr: BigInt = 0x2000, //TODO: pick addr
+  slaveWhere: TLBusWrapperLocation = CBUS
+)
+
+case object ChipIdPinKey extends Field[Option[ChipIdPinParams]](None)
+
+trait CanHavePeripheryChipIdPin { this: BaseSubsystem =>
+  val chip_id_pin = p(ChipIdPinKey).map { params => 
+    val tlbus = locateTLBusWrapper(params.slaveWhere)
+    val device = new SimpleDevice("chip-id-reg", Nil)
+    val id_bits = log2Up(params.numChips)
+
+    val inner_io = tlbus {
+      val node = TLRegisterNode(
+        address = Seq(AddressSet(params.chipIdAddr, 0xfff)), //TODO: fix address set size
```

Comment:
Lets make the address set size 4096 bytes, to match all the other devices.

A wider size than the actual register also simplifies address decoding.